src/app/components/product-list/product-list.component.ts

```diff
@@ -19,45 +19,51 @@ export class ProductListComponent implements OnInit {
   pageLength: number = 0;
   pageSize: number = 6;
   pageSizeOptions: number[] = [this.pageSize, this.pageSize + 4];
-  pageEvent: PageEvent | undefined
+  pageEvent: PageEvent | undefined;
 
   // Products
   products: IProduct[] = [];
   paginationProduct: IProduct[] = [];
 
   public productsCategory: String[] = [];
-  filteredProduct: IProduct[] = []
+  filteredProduct: IProduct[] = [];
 
   constructor(
-    private productService: ProductService) {
+    private productService: ProductService,
+    private cartService: CartService) {
   }
 
   ngOnInit(): void {
     this.productService.getProducts().subscribe(products => {
       this.products = products;
-      this.displayProducts(this.products)
+      this.displayProducts(this.products);
```

Comment:
Po co przekazujesz `this.products` w parametrze do tych funkcji, skoro możesz uzyskać do nich dostęp wewnątrz funkcji bez użwania parametów